lib/task/import/validate/csvDuplicateColumnNameValidator.class.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * CSV Duplicate column name test. Throw error if any column names are repeated in the file.
+ *
+ * @author     Steve Breker <sbreker@artefactual.com>
+ *
+ * @internal
+ * @coversNothing
+ */
+class CsvDuplicateColumnNameValidator extends CsvBaseValidator
+{
+    const TITLE = 'Duplicate Column Name Check';
+    protected $columnFrequency = [];
+
+    public function __construct(array $options = null)
+    {
+        $this->setTitle(self::TITLE);
+
+        parent::__construct($options);
+    }
+
+    public function reset()
+    {
+        $this->columnFrequency = [];
+
+        parent::reset();
+    }
+
+    public function testRow(array $header, array $row)
+    {
+        parent::testRow($header, $row);
+        $header = array_map('trim', $header);
+
+        // Only empty on first iteration.
+        if (empty($this->columnFrequency)) {
+            foreach ($header as $columnName) {
+                if (!array_key_exists($columnName, $this->columnFrequency)) {
```

Comment:
Opinion: I prefer

```
if (!isset($this->columnFrequency[$columnName])) {
```

And it sounds like it might be faster than `array_key_exists()` (though the behaviour is slightly different) based on the comments on https://www.php.net/manual/en/function.array-key-exists.php

Reply:
Addressed in new commit. Replaced uses of `array_key_exists()` with `isset()`.